lib/protobuf/dsl.ex

```diff
@@ -198,10 +218,20 @@ defmodule Protobuf.DSL do
       oneof: Enum.reverse(oneofs),
       enum?: Keyword.get(options, :enum) == true,
       map?: Keyword.get(options, :map) == true,
-      extension_range: extensions
+      extension_range: extensions,
+      unknown_varints_field: gen_unique_varints_field(Map.keys(field_tags))
     }
   end
 
+  defp gen_unique_varints_field(existing_field_names) do
+    indexed_names =
+      Stream.iterate(0, &(&1 + 1))
+      |> Stream.map(&:"__unknown_varints_#{&1}__")
+
+    Stream.concat([:__unknown_varints__], indexed_names)
+    |> Enum.find(fn tentative_name -> tentative_name not in existing_field_names end)
+  end
```

Comment:
I don't think we need to generate a unique name, we have the same issue if someone has a message with `__struct__` field.

Reply:
@ericmj I mean, sure. I would still probably keep the `get_unknown_varints/1` function because I don't want users to access `:__unknown_varints__` though, no? If yes, then I think generating the field and being sure it's unique is not a big deal?

Reply:
Yeah, I think we should still have `get_unknown_varints/1` regardless. No strong opinion on the field name :)